pheno_data/03-dx_extract_dataset_R.qmd

```diff
@@ -0,0 +1,295 @@
+
+```{r}
+#| trusted: true
+# The referenced Dataset is private and provided only to demonstrate an example input. The user will need to supply a permissible and valid record-id
+# Assign project-id of dataset
+# Assign dataset record-id
+rid <- 'record-G406j8j0x8kzxv3G08k64gVV'
+# Assign joint dataset project-id:record-id
+dataset <- rid
+```
+
+### 2. Call “dx extract_dataset” using a supplied dataset
+
+We'll use the `{glue}` package to put our bash commands together for `dx extract_dataset`, and use `system()` to execute our bash code.
+
+`glue::glue()` has the advantage of not needing to `paste()` together strings. The string substitution is cleaner.
+
+```{r}
+#| trusted: true
+cmd <- glue::glue("dx extract_dataset {dataset} -ddd")
+
+cmd
+```
+
+Let's execute our command using `system()` and then we will list the files that result using `list.files()`. We generate three files in the directory in JupyterLab storage:
+
+- *dataset_name*`.codings.csv`
+- *dataset_name*`.data_dictionary.csv`
+- *dataset_name*`.entity_dictionary.csv`
+
+```{r}
+#| trusted: true
+system(cmd)
+list.files()
+```
+
+#### Preview data in the three dictionary (*.csv) files
+
+```{r}
+#| trusted: true
+#codings_file <- system("ls *.codings.csv", intern = TRUE)
+codings_file <- list.files(pattern="*.codings.csv")
+codings_df <- read_csv(codings_file, show_col_types = FALSE)
+head(codings_df)
+```
+
+```{r}
+#| trusted: true
+entity_dict_file <- system("ls *.entity_dictionary.csv", intern=TRUE)
+entity_dict_df <- read_csv(entity_dict_file, show_col_types = FALSE)
+head(entity_dict_df)
+```
+
+### Understanding the Data Dictionary File
+
+The data dictionary is the glue for the entire dataset. It maps:
+
+- Entity to Fields
+- Fields to Codings
+- Entity to Entity
+
+We'll use the data dictionary to understand how to building our list of fields, and later, we'll join it to the codings file to build a list of fields and their coded values.
+
+There are more columns to the data dictionary, but let's first see the `entity`, `name`, `title`, and `type` columns:
+
+```{r}
+#| trusted: true
+#data_dict_file <- system("ls *.data_dictionary.csv", intern=TRUE)
+data_dict_file <- list.files(pattern="*.data_dictionary.csv")
+data_dict_df <- read_csv(data_dict_file, show_col_types = FALSE)
+data_dict_df <- data_dict_df 
+
+data_dict_df %>%
+        select(entity, name, title, type) %>%
+        head()
+```
+
+### 3. Parse returned metadata and extract entity/field names
+
+Let's search for some fields. We want the following fields:
+
+- `Coffee intake | instance 0`
+- `Sex` (Gender)
+- `Smoked cigarette or pipe within last hour | Instance 0`
+
+We can use the `{reactable}` package to make a searchable table of the data dictionary. This will help in finding fields.
+
+Note the search box in the top right of the table - when we have many fields, we can use the search box to find fields of interest. Try searching for `Coffee intake` and see what fields pop up.
```

Comment:
Where is this "top right" referring to? Is this in Rstudio?

Reply:
It's a searchable table - it will only will show up if you execute the code